components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.input.validation.mgt.model.validators;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.math.NumberUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.input.validation.mgt.exceptions.InputValidationMgtClientException;
+import org.wso2.carbon.identity.input.validation.mgt.model.Property;
+import org.wso2.carbon.identity.input.validation.mgt.model.ValidationContext;
+import org.wso2.carbon.identity.input.validation.mgt.model.Validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.MAX_LENGTH;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.MIN_LENGTH;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_DEFAULT_MIN_MAX_MISMATCH;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_PROPERTY_NOT_SUPPORTED;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_PROPERTY_TYPE_MISMATCH;
+
+/**
+ * Abstract rules validator.
+ */
+public abstract class AbstractRulesValidator implements Validator {
+
+    private static final Log log = LogFactory.getLog(AbstractRulesValidator.class);
+
+    @Override
+    public boolean canHandle(String validatorName) {
+
+        return StringUtils.equalsIgnoreCase(validatorName, this.getClass().getSimpleName());
+    }
+
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        return true;
+    }
+
+    @Override
+    public boolean validateProps(ValidationContext context) throws InputValidationMgtClientException {
+
+        Map<String, String> properties = context.getProperties();
+        // Validated supported properties.
+        validatePropertyName(properties, this.getClass().getSimpleName(), context.getTenantDomain());
+
+        if (properties.get(MIN_LENGTH) != null && validatePositiveNumber(properties.get(MIN_LENGTH), MIN_LENGTH,
+                context.getTenantDomain())) {
+            properties.remove(MIN_LENGTH);
```

Comment:
It returns true if it a negative number. Hope we need to change the method name

Reply:
In my opinion, if user provides invalid property attribute in the request body, response have to be 400 bad request (like for un-supporting attributes and validators) rather than reponse with 200 by silently removing invalid property attributes.